### PR TITLE
[Editorial] Stars and Bars

### PR DESCRIPTION
**Pull request by brebenelmihnea**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [ ] I have linked this PR to any issues that it closes.

### CODE REVIEW

**Comment by brebenelmihnea** [comment]
AC proof

https://www.spoj.com/status/MARBLES,mihnea/

image

**Review by envyaims — APPROVED** [review]
i mean.. i would rather you put a more interesting problem for sample rather than literally what stars and bars is but other than that lgtm

File: content/4_Gold/Combinatorics.mdx [hunk 1]
```diff
@@ -923,6 +923,90 @@ int main() {
...
+$$
+||\bigstar \bigstar \bigstar  \\
+|\bigstar |\bigstar \bigstar  \\
+|\bigstar \bigstar| \bigstar  \\
+\bigstar \bigstar \bigstar ||  \\
+$$
+
+As you've probably noticed, there can be empty boxes - when we put all the stars in the first or second box. There may be cases in
+which the all the boxes should be non-empty. In that case, the number of ways to put $n$ indistinguishable objects into $k$
+distinguishable **non-empty** boxes is: $\binom{n-1}{k-1}$
```
Comment (SansPapyrus683) [inline content/4_Gold/Combinatorics.mdx:971]: put this above the problem..

File: content/4_Gold/Combinatorics.mdx [hunk 2]
```diff
@@ -923,6 +923,90 @@ int main() {
...
+
+### Implementation
+
+For this problem we should think the other way around: let's say that the $k$ colors from which we choose are in fact boxes and instead of choosing $n$ marbles we put them
+in the respective boxes. The problem has the restriction that we should pick at least one marble of all kinds, which means in our new perspective that all the boxes should be
+non-empty. Thus, the answer is obtained by the second formula: $\binom{n-1}{k-1}$
```
Comment (SansPapyrus683) [inline content/4_Gold/Combinatorics.mdx:977]: and this in the explanation header

File: content/4_Gold/Combinatorics.mdx [hunk 3]
```diff
@@ -923,6 +923,90 @@ int main() {
...
+#include <iostream>
+
+using namespace std;
+
+int main() {
+	auto comb = [](int n, int k) -> long long {
+		if (k > n - k) k = n - k;
+		long long ans = 1;
+		for (int i = 0; i < k; i++) { ans = ans * (n - i) / (i + 1); }
+		return ans;
```
Comment (SansPapyrus683) [inline content/4_Gold/Combinatorics.mdx:994]: just make this its own function

File: content/4_Gold/Combinatorics.mdx [hunk 4]
```diff
@@ -923,6 +923,90 @@ int main() {
...
+	};
+
+	int T;
+	cin >> T;
```
Comment (SansPapyrus683) [inline content/4_Gold/Combinatorics.mdx:998]: rename to `test_num`

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: content/4_Gold/Combinatorics.mdx (hunk 4 above)
Comment (envyaims) [reply content/4_Gold/Combinatorics.mdx:998]: 💀

File: content/4_Gold/Combinatorics.mdx (hunk 1 above)
Comment (brebenelmihnea) [reply content/4_Gold/Combinatorics.mdx:967]: done

File: content/4_Gold/Combinatorics.mdx (hunk 3 above)
Comment (brebenelmihnea) [reply content/4_Gold/Combinatorics.mdx]: done

File: content/4_Gold/Combinatorics.mdx (hunk 2 above)
Comment (brebenelmihnea) [reply content/4_Gold/Combinatorics.mdx:975]: ok

File: content/4_Gold/Combinatorics.mdx (hunk 4 above)
Comment (brebenelmihnea) [reply content/4_Gold/Combinatorics.mdx]: done

File: content/4_Gold/Combinatorics.mdx [hunk 5]
```diff
@@ -923,6 +923,90 @@ int main() {
...
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <iostream>
+
+using namespace std;
+
+long long comb(int n, int k) {
+	if (k > n - k) k = n - k;
```
Comment (ryanchou-dev) [inline content/4_Gold/Combinatorics.mdx:990]: braces

**Review by ryanchou-dev — APPROVED** [review]
like envy said, would recommend a more interesting problem

File: content/4_Gold/Combinatorics.mdx (hunk 5 above)
Comment (brebenelmihnea) [reply content/4_Gold/Combinatorics.mdx]: done

**Review by SansPapyrus683 — APPROVED** [review]
formatting seems fine lol

**Pull request merged**